backend/platform_settings_v2/platform_auth_service.py

```diff
@@ -0,0 +1,253 @@
+            result[PlatformServiceConstants.ID] = platform_key.id
+            result[PlatformServiceConstants.KEY_NAME] = platform_key.key_name
+            result[PlatformServiceConstants.KEY] = platform_key.key
+
+            logger.info(f"platform_key is generated for {organization.id}")
+            return result
+        except IntegrityError as error:
+            logger.error(f"Integrity error - failed to generate platform key : {error}")
+            raise DuplicateData(
+                f"{ErrorMessage.KEY_EXIST}, \
+                            {ErrorMessage.DUPLICATE_API}"
+            )
+        except Exception as error:
+            logger.error(f"Failed to generate platform key : {error}")
+            raise InternalServiceError()
+
+    @staticmethod
+    def delete_platform_key(id: str) -> None:
+        """This is a delete operation. Use this function only if you know what
+        you are doing.
+
+        Args:
+            id (str): _description_
+
+        Raises:
+            error: _description_
+        """
+        try:
+            platform_key: PlatformKey = PlatformKey.objects.get(pk=id)
+            platform_key.delete()
+            logger.info(f"platform_key {id} is deleted")
+        except IntegrityError as error:
+            logger.error(f"Failed to delete platform key : {error}")
+            raise DuplicateData(
+                f"{ErrorMessage.KEY_EXIST}, \
+                            {ErrorMessage.DUPLICATE_API}"
+            )
+        except Exception as error:
+            logger.error(f"Failed to delete platform key : {error}")
+            raise InternalServiceError()
+
+    @staticmethod
+    def refresh_platform_key(id: str, user: User) -> dict[str, Any]:
+        """Method to refresh a platform key.
+
+        Args:
+            id (str): Unique id of the key to be refreshed
+            new_key (str): Value to be updated.
+
+        Raises:
+            error: IntegrityError
+        """
+        try:
+            result: dict[str, Any] = {}
+            platform_key: PlatformKey = PlatformKey.objects.get(pk=id)
+            platform_key.key = str(uuid.uuid4())
+            platform_key.modified_by = user
+            platform_key.save()
+            result[PlatformServiceConstants.ID] = platform_key.id
+            result[PlatformServiceConstants.KEY_NAME] = platform_key.key_name
+            result[PlatformServiceConstants.KEY] = platform_key.key
+
+            logger.info(f"platform_key {id} is updated by user {user.id}")
+            return result
+        except IntegrityError as error:
+            logger.error(f"Integrity error - failed to refresh platform key : {error}")
+            raise DuplicateData(
+                f"{ErrorMessage.KEY_EXIST}, \
+                            {ErrorMessage.DUPLICATE_API}"
+            )
+        except Exception as error:
+            logger.error(f"Failed to refresh platform key : {error}")
+            raise InternalServiceError()
```

Comment:
May be add more contexts in to the error messages to figure out which org or platform_key id etc,.

Reply:
Removed unwanted generic excpetion handler